src/App.tsx

```diff
@@ -1,14 +1,83 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [filter, setFilter] = useState('all');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [activeTodo, setActiveTodo] = useState<Todo | undefined>(undefined);
+  const [query, setQuery] = useState('');
+  const [loadingTodos, setLoadingTodos] = useState(true);
+
+  useEffect(() => {
+    setLoadingTodos(true);
+
+    switch (filter) {
+      case 'all':
+        getTodos()
+          .then(allTodos =>
+            allTodos.filter(todo =>
+              todo.title.toLowerCase().includes(query.toLowerCase()),
+            ),
+          )
+          .then(setTodos)
+          .finally(() => setLoadingTodos(false));
+        break;
+      case 'active':
+        getTodos()
+          .then(allTodos =>
+            allTodos.filter(
+              todo =>
+                !todo.completed &&
+                todo.title.toLowerCase().includes(query.toLowerCase()),
+            ),
+          )
+          .then(setTodos)
+          .finally(() => setLoadingTodos(false));
+        break;
+      case 'completed':
+        getTodos()
+          .then(allTodos =>
+            allTodos.filter(
+              todo =>
+                todo.completed &&
+                todo.title.toLowerCase().includes(query.toLowerCase()),
+            ),
+          )
+          .then(setTodos)
+          .finally(() => setLoadingTodos(false));
+        break;
+      default:
+        setTodos([]);
+        setLoadingTodos(false);
+        break;
+    }
+  }, [filter, query]);
+
+  const handleFilter = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setFilter(event.target.value);
+  };
+
+  const handleModalOpen = (activeTodoItem: Todo | undefined) => {
```

Comment:
this wrapper is redundant - just use `setActiveTodo` instead of `handleModalOpen` everywhere

Reply:
oh, yes, you are correct, removed this.